src/coingecko/Coingecko.ts

```diff
@@ -113,15 +115,28 @@ export class Coingecko {
 
   async call(path: string) {
     const sendRequest = async () => {
+      const { host, proHost } = this;
+
+      // Try basic, and then pro as fallback.
+      this.logger.debug({ at: "sdk-v2/coingecko", message: `Sending GET request to host ${host}` });
+      const basicUrl = `${host}/${path}`;
+
       try {
-        const { host } = this;
-        const url = `${host}/${path}`;
-        const result = await axios(url, { params: { x_cg_pro_api_key: this.apiKey } });
-        this.logger.debug({ at: "sdk-v2/coingecko", message: `Sent GET request to url ${result.request.responseURL}` });
+        const result = await axios(basicUrl, { timeout: 200 });
         return result.data;
       } catch (err) {
-        const msg = get(err, "response.data.error", get(err, "response.statusText", "Unknown Coingecko Error"));
-        throw new Error(msg);
+        this.logger.debug({
+          at: "sdk-v2/coingecko",
+          message: `Basic CG url request failed, falling back to CG PRO host ${proHost}`,
+        });
+        const proUrl = `${proHost}/${path}`;
```

Comment:
It's not guaranteed that apiKey is defined, which would cause a subsequent rejection from the endpoint and would falsely report to the caller that the price lookup failed due to an auth/no-key error.

Reply:
Great point! added logic to only use bbasic api if no pro api key